webapp/sources/rudder/rudder-web/src/main/scala/com/normation/rudder/web/snippet/administration/PropertiesManagement.scala

```diff
@@ -483,58 +485,186 @@ class PropertiesManagement extends DispatchSnippet with Loggable {
+    def checkSyslogProtocol(input : String) : JsCmd = {
+      SyslogProtocol.parse(input).toBox match {
+        case Full(newValue) =>
+          syslogProtocol = newValue
+          Noop
+        case eb: EmptyBox =>
+          Noop
+      }
+    }
+
+    def displayDisableSyslogSectionJS(input: String): JsCmd = {
+      if (input == httpsLegacySupport) {
+        reportProtocol = AgentReportingHTTPS
+        disabledSyslog = false
+        JsRaw(""" $('#syslogProtocol').show(); """)
+      } else {
+        AgentReportingProtocol.parse(input).toBox match {
+          case Full(protocol) =>
+            reportProtocol = protocol
+            protocol match {
+              case AgentReportingHTTPS =>
+
+                disabledSyslog = true
+                JsRaw(""" $('#syslogProtocol').hide(); """)
+              case AgentReportingSyslog =>
+
+                disabledSyslog = false
+                JsRaw(""" $('#syslogProtocol').show(); """)
+            }
+          case eb: EmptyBox =>
+            Noop
         }
       }
+    }
 
-      val checkboxInitValue = initReportsProtocol == SyslogUDP
+    val zip : Box[(AgentReportingProtocol,Boolean)] = (initReportProtocol,initDisabledSyslog) match {
+      case (Full(a),Full(b)) => Full((a,b))
+      case (eb1 : EmptyBox, eb2: EmptyBox) =>
+        val fail1 = eb1 ?~! "Error when getting reporting protocol"
+        val fail2 = eb2 ?~! "Error when getting disabled syslog protocol"
+        Failure(s"Error when fetch reporting protocol options: ${fail1.messageChain} - ${fail2.messageChain}")
+      case (eb1 : EmptyBox, _) =>
+        eb1 ?~! "Error when getting reporting protocol"
+      case (_, eb2 : EmptyBox) =>
+        eb2 ?~! "Error when getting disabled syslog protocol"
+    }
+    def labelAndValue(protocol : AgentReportingProtocol, disabledSyslog : Boolean) =
+      protocol match {
+        case AgentReportingHTTPS if ! disabledSyslog =>
+          ( protocol.value +" with syslog support for migration purpose (old agent or legacy systems)",  httpsLegacySupport)
+        case _ => ( protocol.value + " only",  protocol.value)
+      }
 
-     ( "#reportProtocol" #> {
-          SHtml.ajaxCheckbox(
-              checkboxInitValue
-            , (newValue : Boolean) => {
-                reportProtocol = if (newValue) SyslogUDP else SyslogTCP
-                check
+      (
+        "#reportProtocol" #> {
+          zip match {
+            case Full((initReport, initDisabled)) =>
+              val (_,initValue) = labelAndValue(initReport,initDisabled)
+              def radioHtml(protocol : AgentReportingProtocol, disabledSyslog : Boolean): NodeSeq = {
+                val (label,value) = labelAndValue(protocol,disabledSyslog)
+                val inputId = value + "-id"
+                val ajaxCall = SHtml.ajaxCall(Str(""), _ => displayDisableSyslogSectionJS(value) & check)._2.toJsCmd
+                val inputCheck = if (initReport == protocol && initDisabled == disabledSyslog) {
```

Comment:
why the condition on disabledSyslog ?

Reply:
(or on protocol - i don't get why there is both)

Reply:
Because we manage both properties with that input ... it' a three state select